### PR TITLE
Close sqlite database when user quits

### LINKED ISSUE #2178: SQLite test may fail on CI
https://github.com/thelounge/thelounge/blob/5aa9d7e1dc68ece7edda8742b248e22206107006/test/plugins/sqlite.js#L36-L39

Some kind of race condition is happening and the file is not existing after calling `enable`. Is sqlite doing some async-ness internally that would cause this?

Perhaps using `database.serialize` could be a solution here.

### PR DESCRIPTION
**Pull request by xPaw**
.. or on server shutdown.

Hopefully fixes #2178 by moving file exists check into `after`.

### CODE REVIEW

File: src/plugins/sqlite.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ class MessageStorage {
 
 		this.isEnabled = true;
 
-		this.database = new sqlite3.cached.Database(sqlitePath);
```
Comment (xPaw) [inline src/plugins/sqlite.js]: Caching doesn't do anything for us. (all it does is keeps internal reference for the filename).

File: test/plugins/sqlite.js [hunk 2]
```diff
@@ -22,25 +22,28 @@ describe("SQLite Message Storage", function() {
 		}
 	});
 
+	after(function(done) {
+		store.close((err) => {
+			expect(err).to.be.null;
+			expect(fs.existsSync(expectedPath)).to.be.true;
```
Comment (astorije) [inline test/plugins/sqlite.js]: Expects in after? This is becoming increasingly bizarre lol. We're not really supposed to do that

Comment (xPaw) [reply test/plugins/sqlite.js]: `after` can be a test just like any other, no? we're just guaranteeing the order here.

Comment (astorije) [reply test/plugins/sqlite.js]: I've noticed in the past that the reporter usually lose it when there is a test in a before/after hook.
Because tests in this file **need** to run in order anyway, and this effectively runs tests, I would move it in a `it` statement at the very bottom of the file.

Comment (astorije) [reply test/plugins/sqlite.js]: Does `expect(fs.existsSync(expectedPath)).to.be.true;` test that the file wasn't destroyed when exiting or something? Do we want/need to test the file is valid, and/or not empty, and/or load, etc.?

Comment (xPaw) [reply test/plugins/sqlite.js]: It's testing that the file is created at the expected path.

Comment (astorije) [reply test/plugins/sqlite.js]: Ah, makes sense. Then in an `it` at the end should do, no?

Comment (xPaw) [reply test/plugins/sqlite.js]: Done.
